lib/surface/compiler.ex

```diff
@@ -981,7 +970,7 @@ defmodule Surface.Compiler do
       Hint: replace `<slot>` with `<#slot>`
       """
 
-      IOHelper.warn(message, compile_meta.caller, &(&1 + line))
+      IOHelper.warn(message, compile_meta.caller, fn _ -> line end)
```

Comment:
i think that makes sense - i actually did make changes to just pass in a line instead of a function, but this had already grown quite a bit. i'll do those as a separate PR after this one